content/docs/rfcs/64/README.md

```diff
@@ -0,0 +1,258 @@
+2. [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) to allow other peers to request historical messages from this node. The store SHOULD be configured to retain at least `12` hours of messages per supported shard.
+3. [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/spec/19/) to allow resource-restricted peers to request publishing a message to the network on their behalf.
+4. [34/WAKU2-PEER-EXCHANGE](https://rfc.vac.dev/spec/34/) to allow resource-restricted peers to discover more peers in a resource efficient way.
+
+#### Non-relay nodes
+
+Nodes MAY opt out of relay functionality on any network shard
+and instead request services from relay nodes as clients
+using any of the defined service protocols:
+1. [12/WAKU2-FILTER](https://rfc.vac.dev/spec/12/) to subscribe to messages matching a specific content filter.
+2. [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) to request historical messages matching a specific content filter.
+3. [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/spec/19/) to request publishing a message to the network.
+4. [34/WAKU2-PEER-EXCHANGE](https://rfc.vac.dev/spec/34/) to discover more peers in a resource efficient way.
+
+### Applications
+
+Applications are the higher-layer projects or platforms that make use of the generalized messaging capability of the network.
+In other words, an application defines a payload used in the various [10/WAKU2](https://rfc.vac.dev/spec/10/) protocols.
+Any participant in an application SHOULD make use of an underlying node in order to communicate on the network.
+Applications SHOULD make use of an [autosharding](#autosharding) API
+to allow the underlying node to automatically select the target shard on the Waku Network. See the section on [autosharding](#autosharding) for more.
+
+## RLN rate-limiting
+
+The [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/) network uses [32/RLN-V1](https://rfc.vac.dev/spec/32/) proofs
+to ensure that a pre-agreed rate limit is not exceeded by any publisher.
+While the network is under capacity,
+individual relayers MAY choose to freely route messages without RLN proofs
+up to a discretionary bandwidth limit after which messages without proofs MUST be discarded.
+This implies that quality of service and reliability is significantly lower for messages without proofs
+and at times of high network utilization these messages may not be relayed at all.
+
+### RLN Parameters
+
+For the Waku Network,
+the `epoch` is set to `1` second
+and the maximum number of messages published per `epoch` is limited to `1` per publisher.
+The `max_epoch_gap` is set to `20` seconds,
+meaning that validators MUST _ignore_ messages with an `epoch` more than 20 seconds into the past or future compared to the validator's own clock.
+
+### Memberships
+
+Each publisher to the Waku Network SHOULD register an RLN membership
+with contract (TBD)[contract address]
```

Comment:
The latest contract used is https://github.com/waku-org/waku-rln-contract/commit/19fded82bca07e7b535b429dc507cfb83f10dfcf and deployed to sepolia - the question I have is if we should mention that it's unaudited, but the rln circuits are.

Reply:
Yes, I think it's good to mention that. May also want to define here that it's a _registry_ contract and on sepolia. Since this is a raw spec we can freely update if we deploy to mainnet.

Reply:
Mmm. Ended up not saying anything about this as it falls a bit outside spec scope. I added some sections in [fe052be](https://github.com/vacp2p/rfc/pull/616/commits/fe052be02352dd650b076e862cb1bd63bc9c310b). Feel free to take a look and let me know what you think!